data-prepper-plugins/kafka-source/src/main/java/org/opensearch/dataprepper/plugins/kafka/source/KafkaSource.java

```diff
@@ -0,0 +1,126 @@
+  }
+  
+  @Override
+  public void start(Buffer<Record<Object>> buffer) {
+    sourceConfig.getTopics().forEach(topic -> {
+      Properties consumerProperties = getConsumerProperties(topic);
+        try {
+        	workers = workers + topic.getTopic().getConsumerGroupConfig().getWorkers(); 
+        	consumerGroups = topic.getTopic().getConsumerGroupConfig().getGroupName();
+            IntStream.range(0, workers).forEach(index -> {
+              String consumerId = Integer.toString(index + 1);
+              multithreadedConsumer = new MultithreadedConsumer(consumerId,
+                      consumerGroups, consumerProperties, topic.getTopic(), buffer, pluginMetrics);
+              consumers.add(multithreadedConsumer);
+            });
+        } catch (Exception e) {
+          LOG.error("Failed to setup the Kafka Source Plugin.", e);
+          kafkaWorkerThreadProcessingErrors.increment();
+        }
+    });
+      executorService = Executors.newFixedThreadPool(workers);
+      try {
+         executorService.invokeAll(consumers);
+      } catch (InterruptedException e) {
+        LOG.error("Kafka source executor service is failed to invoke thread-pool...");
+        Thread.currentThread().interrupt();
+      }
+  }
+
+  @Override
+  public void stop() {
+   LOG.info("Shutting down Consumers...");
+   executorService.shutdown();
+   try {
+      if (!executorService.awaitTermination(
+          sourceConfig.getTopics().get(0).getTopic().getConsumerGroupConfig().getThreadWaitingTime().toSeconds(), TimeUnit.MILLISECONDS)) {
+    	  LOG.info("Consumer threads are waiting for shutting down...");
+        executorService.shutdownNow();
+      }
+    } catch (InterruptedException e) {
+      if (e.getCause() instanceof InterruptedException) {
+        LOG.error("Interrupted during consumer shutdown, exiting uncleanly...");
+        executorService.shutdownNow();
+        Thread.currentThread().interrupt();
+      }
+    }
+    LOG.info("Consumer shutdown successfully...");
+  }
+
+  private Properties getConsumerProperties(TopicsConfig topicConfig) {
+    Properties properties = new Properties();
+    try {
+      properties.put(ConsumerConfig.AUTO_COMMIT_INTERVAL_MS_CONFIG,
+    		  topicConfig.getTopic().getConsumerGroupConfig().getAutoCommitInterval().toSecondsPart());
+      properties.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG,
+    		  topicConfig.getTopic().getConsumerGroupConfig().getAutoOffsetReset());
+      properties.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, sourceConfig.getBootStrapServers());
+      properties.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG,
+    		  topicConfig.getTopic().getConsumerGroupConfig().getAutoCommit());
+      properties.put(ConsumerConfig.GROUP_ID_CONFIG, topicConfig.getTopic().getConsumerGroupConfig().getGroupId());
+      setPropertiesForSchemaType(topicConfig, properties, topicConfig.getTopic().getSchemaConfig().getSchemaType());
```

Comment:
we should validate all these inputs if there are within in expected range, unless there are  validated elsewhere